altlas/Assets/scripts/MapGenerator.cs

```diff
@@ -290,11 +291,65 @@ public void spawnStacksInRow(float horLength, float verLength, List<List<MapData
         }
     }
 
-    /**
-     * spawnStackInRow with fixed drawer length
-     * */
-    public void spawnRowInDrawer(MapData[][] stacks, string category) {
-        spawnStacksInRow(DRAWER_HOR_LEN, DRAWER_VER_LEN, stacks, category);
+    void setLabelOnDraw(KeyValuePair<string, Dictionary<string, List<MapData>>> category)
+    {     
+        GameObject go = new GameObject("myText");
+        go.AddComponent<TextMesh>();
+        TextMesh tm = go.GetComponent(typeof(TextMesh)) as TextMesh;
+        go.name = category.Key + "_label";
+        var drawer = getDrawerObjectFromCategory(category.Key);
+        go.transform.parent = drawer.transform;
+
+        tm.text = categoryText[category.Key];
+        tm.anchor = TextAnchor.UpperCenter;
+        tm.alignment = TextAlignment.Center;
+        tm.transform.localScale = drawer.transform.Find("knob").transform.localScale;
```

Comment:
what.
just set this to a magic number, don't add random references to other objects